### PR TITLE
Lodash: Remove completely from `@wordpress/format-library` package

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR removes all of the Lodash from the `@wordpress/format-library` package, including the `lodash` dependency altogether. There are just a few usages and conversion is pretty straightforward. 

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

We're replacing a few usages with their native counterparts. 

## Testing Instructions

* On both web and mobile, make sure to test: 
  * Setting text color in a paragraph block (see #36028 for testing details on mobile)
  * Setting a text to be a link in a paragraph block (see #12249 for testing details on mobile)
* Verify all checks are green.

### CODE REVIEW

**Review by aristath — APPROVED** [review]
I didn't test on mobile, but tested on desktop and reviewed the code. LGTM! 👍

**Comment by tyxla** [comment]
> I didn't test on mobile, but tested on desktop and reviewed the code. LGTM! 👍

Thanks, @aristath! 

I'll hold off merging until the React Native folks to confirm this isn't breaking anything on mobile.

File: packages/format-library/src/link/index.native.js
Comment (dcalhoun) [inline packages/format-library/src/link/index.native.js]: This change causes a crash. It appears `value.formats[ startIndex ]` sometimes returns `undefined`, which `find` must've gracefully handled previously. 

```suggestion
						value.formats[ startIndex ]?.find(
```

File: packages/format-library/src/link/index.native.js
Comment (dcalhoun) [inline packages/format-library/src/link/index.native.js]: Similar to the `startIndex`, this can be `undefined`. 
```suggestion
						value.formats[ endIndex ]?.find(
```

**Review by dcalhoun — CHANGES_REQUESTED** [review]
Thank you for notifying the mobile team. 🙇🏻 

I tested these changes on an iPhone simulator and Android emulator. I encountered one crash that we should resolve before merging this work please.

File: packages/format-library/src/link/index.native.js
Comment (tyxla) [reply packages/format-library/src/link/index.native.js]: Thanks for this! Addressed in 6e7ec5fe30b2db3aa06859ae115fda878a6f8cea.

File: packages/format-library/src/link/index.native.js
Comment (tyxla) [reply packages/format-library/src/link/index.native.js]: Addressed as part of 6e7ec5fe30b2db3aa06859ae115fda878a6f8cea.

**Comment by tyxla** [comment]
Feedback has been addressed - could I get one last look, @dcalhoun?

**Review by dcalhoun — APPROVED** [review]
🚀 Great work!

**Comment by tyxla** [comment]
Thank you, @dcalhoun 🙌

**Pull request merged**
